tests/csapi/apidoc_content_test.go

```diff
@@ -0,0 +1,63 @@
+package csapi_tests
+
+import (
+	"io/ioutil"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+)
+
+var contentID *url.URL
+var contentText = "Here is the content I am uploading"
+
+func TestContent(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	authedClient := deployment.Client(t, "hs1", "@alice:hs1")
+	unauthedClient := deployment.Client(t, "hs1", "")
+	// sytest: POST /media/r0/upload can create an upload
+	t.Run("POST /media/r0/upload can create an upload", func(t *testing.T) {
+
+		gotUrl := ""
+		queryParams := url.Values{}
+		queryParams.Set("access_token", authedClient.AccessToken)
+
+		contentType := client.WithContentType("text/plain")
+		reqBody := client.WithRawBody([]byte(contentText))
+		res := unauthedClient.MustDoFunc(t, "POST", []string{"_matrix", "media", "r0", "upload"}, reqBody, contentType, client.WithQueries(queryParams))
+
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			t.Fatalf("unable to read response body: %v", err)
+		}
+
+		gotUrl = gjson.GetBytes(body, "content_uri").Str
+		if gotUrl == "" {
+			t.Fatal("expected URL to hae propogated the value")
```

Comment:
This isn't a helpful error message because it doesn't explain where this URL came from (the error) and give context to other fields that are in the JSON. Imagine if `body` had the shape:
```json
{
    "content_url": "mxc://foo/bar"
}
```
This looks correct but it isn't. Running this test would fail with:
```
expected URL to hae propogated the value
```
What does this mean from the perspective of the person testing their code? Not a lot. An improvement would be:
```
expected 'content_uri' in response body
```
This lets the developer know that a key is missing. We can be even more helpful though by providing context: what keys were in the response? If we don't, the developer may spend 20-30 minutes thinking that there is no URI at all in the response which isn't quite true. The best message would be:
```
expected 'content_uri' in response body, got: { "content_url": "mxc://foo/bar" }
```
This _immediately_ tells the developer that they made a typo: `url` instead of `uri`. Good error messages _saves developer time_, please treat them with just as much importance as the actual mechanics of the test.